src/main/java/com/puppycrawl/tools/checkstyle/JavaAstVisitor.java

```diff
@@ -103,6 +103,16 @@ public final class JavaAstVisitor extends JavaLanguageParserBaseVisitor<DetailAs
     /** String representation of the right shift operator. */
     private static final String RIGHT_SHIFT = ">>";
 
+    /**
+     * The tokens here are technically expressions, but should
+     * not return an EXPR token as their root.
+     */
+    private static final int[] EXPRESSIONS_WITH_NO_EXPR_ROOT = {
```

Comment:
Made this a static field so that we do not have to create array with each method call.